csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Numerics.Tensors;
+using System.Buffers;
+using System.Collections;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    public interface IDisposableReadOnlyCollection<T>: IReadOnlyCollection<T>, IDisposable
```

Comment:
Does the order of the results have any semantic meaning/value? Does the `[0]` element align with the `0`th element in a "metadata" description?

If so - should this be `IReadOnlyList` instead of `IReadOnlyCollection`?  Lists have ordering, so you can index based by an integer.

Reply:
typically there is 1-1 correspondence of the position of the results. However, in some part of the API, one can request for a subset of the named results, in that case there may not be 1-1 correspondence with metadata. Thus keeping it collection.